frontend/src/components/explorer/gemBrowser/GemSearch.vue

```diff
@@ -189,16 +188,16 @@ export default {
       }
     },
     async searchDebounce(searchTerm) {
+      this.$store.dispatch('search/setSearchTermString', searchTerm);
       this.noResult = false;
       this.showSearchCharAlert = searchTerm.length === 1;
-      this.$store.dispatch('search/setSearchTermString', searchTerm);
 
       const canSearch = searchTerm.length > 1;
 
       this.showLoader = canSearch;
       this.showResults = canSearch;
       if (canSearch) {
-        await this.search(searchTerm);
+        await this.search();
```

Comment:
This might have originated when users would enter the search page by url, like it is the case for the global search, but I think these searches are distinct now.